lib/elixir/lib/module.ex

```diff
@@ -1669,16 +1669,20 @@ defmodule Module do
     end
 
     case :ets.lookup(set, key) do
-      [{^key, _, :accumulate}] ->
+      [{_, _, :accumulate}] ->
         :ets.insert(bag, {{:accumulate, key}, value})
 
-      [{^key, {line, <<_::binary>>}, _unread_line}]
+      [{_, {line, <<_::binary>>}, _unread_line}]
       when key in [:doc, :typedoc, :moduledoc] and is_list(stack) ->
         IO.warn("redefining @#{key} attribute previously set at line #{line}", stack)
         :ets.insert(set, {key, value, unread_line})
 
-      _ ->
+      [_] ->
```

Comment:
What do you think about doing a full lookup only if the key is one of those when we care? Otherwise just do a `lookup_element` on the third element? If the value is large, this should probably improve performance significantly - no need to copy it each time it's overridden.

Reply:
I like this. I will play with it.